src/index.js

```diff
@@ -1,66 +1,247 @@
+} catch (error) {
+  console.log(error.message);
+}
+
+// 2
+function parseDateIfValid(dateString) {
+  const date = new Date(dateString);
+
+  if (isNaN(date.getTime())) {
+    throw new Error('Invalid date');
+  }
+
+  return date;
+}
+
+try {
+  console.log(parseDateIfValid('today'));
+} catch (error) {
+  console.error(error.message);
+}
+
+// 3
+function evaluateScript(scriptString) {
+  try {
+    return eval(scriptString);
+  } catch {
+    return null;
+  }
+}
+
+console.log(evaluateScript('1 + 5'));
+
+// 4
+function getUserName(usersID) {
+  return fetch(`https://jsonplaceholder.typicode.com/users/${usersID}`)
+    .then((response) => {
+      if (!response.ok) {
+        return null;
+      }
+      return response.json();
+    })
+    .then((user) => {
+      if (user) {
+        return user.name;
+      }
+      return null;
+    })
+    .catch(() => null);
+}
+getUserName(2).then((name) => console.log(name));
+
+async function getUserNameAsync(usersID) {
+  try {
+    const response = await fetch(
+      `https://jsonplaceholder.typicode.com/users/${usersID}`,
+    );
+    if (!response.ok) {
+      return null;
     }
-    if (secondButton) {
-      secondButton.addEventListener('click', function () {
-        resolve(secondButton);
-      });
+    const user = await response.json();
+    return user.name || null;
+  } catch {
+    return null;
+  }
+}
+getUserNameAsync(1).then((name) => console.log(name));
+
+// 5
+function getAllPostsFromUser(userId) {
+  return new Promise((resolve, reject) => {
+    fetch('https://jsonplaceholder.typicode.com/posts')
+      .then((response) => response.json())
+      .then((posts) => {
+        const userPosts = posts.filter((post) => post.userId === userId);
+        if (userPosts.length > 0) {
+          resolve(userPosts);
+        } else {
+          reject(new Error(`No posts found for user with ID ${userId}`));
+        }
+      })
+      .catch((error) => reject(error));
+  });
+}
+
+getAllPostsFromUser(5)
+  .then((posts) => console.log(posts))
+  .catch((error) => console.error(error));
+
+async function getAllPostsFromUserAsync(userId) {
+  try {
+    const response = await fetch('https://jsonplaceholder.typicode.com/posts');
+    const posts = await response.json();
+    const userPosts = posts.filter((post) => post.userId === userId);
+    if (userPosts.length > 0) {
+      return userPosts;
     }
+  } catch {
+    throw new Error(`No posts found for user with ID ${userId}`);
+  }
+}
+
+getAllPostsFromUserAsync(1)
+  .then((posts) => console.log(posts))
+  .catch((error) => console.error(error));
+
+// 6
+function getAlbumsWithPhotos() {
+  return Promise.all([
+    fetch('https://jsonplaceholder.typicode.com/albums').then((response) =>
+      response.json(),
+    ),
+    fetch('https://jsonplaceholder.typicode.com/photos').then((response) =>
+      response.json(),
+    ),
+  ]).then(([albums, photos]) => {
+    const photosByAlbum = photos.reduce((accumulator, photo) => {
```

Comment:
Good job. However, try figuring out a name different from `accumulator` that's a bit more descriptive.